lib/suggest.js

```diff
@@ -89,169 +91,33 @@ function githubStars (user, repo) {
-                + '&client_secret=' + serverSecrets.gh_client_secret;
-            }
-            request(apiUrl, { headers: { 'User-Agent': 'Shields.io' } }, function(err, res, buffer) {
-              if (err != null) { resolve(null); return; }
-              try {
-                if ((+res.headers['x-ratelimit-remaining']) === 0) { resolve(null); return; }
-                const data = JSON.parse(buffer);
-                const treeArray = data.tree;
-                let licenseBlob;
-                let licenseFilename;
-                // Crawl each file in the root directory
-                for (let i = 0; i < treeArray.length; i++) {
-                  if (treeArray[i].type !== 'blob') {
-                    continue;
-                  }
-                  if (treeArray[i].path.match(/(LICENSE|COPYING|COPYRIGHT).*/i)) {
-                    licenseBlob = treeArray[i].sha;
-                    licenseFilename = treeArray[i].path;
-                    break;
-                  }
-                }
-                // Could not find license file
-                if (!licenseBlob) { resolve(null); return; }
-
-                // Step 4: Get the license blob.
-                let apiUrl = 'https://api.github.com/repos/' + user + '/' + repo + '/git/blobs/' + licenseBlob;
-                const link = 'https://raw.githubusercontent.com/' +
-                  [user, repo, defaultBranch, licenseFilename].join('/');
-
-                if (serverSecrets) {
-                  apiUrl += '?client_id=' + serverSecrets.gh_client_id
-                    + '&client_secret=' + serverSecrets.gh_client_secret;
-                }
-                // Get the raw blob instead of JSON
-                // https://developer.github.com/v3/media/
-                request(apiUrl, { headers: { 'User-Agent': 'Shields.io', 'Accept': 'appplication/vnd.github.raw' } },
-                function(err, res, buffer) {
-                  if (err != null) { resolve(null); return; }
-                  try {
-                    if ((+res.headers['x-ratelimit-remaining']) === 0) { resolve(null); return; }
-                    const license = guessLicense(buffer);
-                    if (license) {
-                      badgeData.text[1] = license;
-                      resolve({
-                        link: link,
-                        badge: shieldsBadge(badgeData),
-                        name: 'GitHub license'
-                      });
-                      return;
-                    } else {
-                      // Not a recognized license
-                      resolve(null);
-                      return;
-                    }
-                  } catch(e) { reject(e); }
-                });
-              } catch(e) { reject(e); }
-            });
-          } catch(e) { reject(e); }
-        });
-      } catch(e) { reject(e); }
-    });
+      } catch(e) {
+        resolve(null);
+      }
+    }, { 'Accept' : 'application/vnd.github.drax-preview+json' })
```

Comment:
So the idea is to suggest the license badge, but only if the badge will show something?

Assuming the badge shows something like "unknown" if the license isn't specified / detectable, I wonder if it's worth suggesting that badge anyway? It could trigger the developer to correctly indicate the license in their repo.

Reply:
I need to access License API to URL to license file. But you're right, we can always suggest license badge. I like this idea. I will fix it.